### DISCUSSION TITLE
Queries with different root nodes but similar contents

### DISCUSSION THREAD

**Discussion by guillaumebrunerie** [tree-sitter/tree-sitter#4701] [Q&A] [answered]
I might be missing something obvious, but I can't figure out how to consolidate several queries where the only difference is in the first node. For instance I currently have something like
```lisp
(assignment_expression
  (identifier) @variable)

(augmented_assignment_expression
  (identifier) @variable)

(update_expression
  (identifier) @variable)
```
But I would rather not repeat myself and write something like this (not working)
```
([assignment_expression augmented_assignment_expression update_expression]
  (identifier) @variable)
```
Is there a way to do it?

**Comment by ObserverOfTime** [comment, marked as answer]
Add a [supertype](https://tree-sitter.github.io/tree-sitter/creating-parsers/2-the-grammar-dsl.html#:~:text=supertypes) that contains those nodes and use that.
```tsq
(expression/identifier) @variable
```
